ecommers/lib/core/provider_models/product_page_provider_model.dart

```diff
@@ -17,10 +18,26 @@ class ProductPageProviderModel extends ProviderModelBase {
           ?.map((url) => CarouselImage(tag: _productModel.id, path: url))
           ?.toList();
     }
+
+    _saveRecentProduct();
   }
 
   void updateImages(List<CarouselImage> assets) {
     images = assets;
     notifyListeners();
   }
+
+  Future _saveRecentProduct() async {
```

Comment:
guess we don't need to write overhead with isBusy ...

let this method to execute: productService.trySaveRecentProduct(
          _productModel.toJson(),
        );